package-testing/android-relay/app/src/main/java/cloud/eppo/android/sdkrelay/TestClientActivity.kt

```diff
@@ -0,0 +1,279 @@
+        .host(EPPO_API_ADDRESS)
+        .isGracefulMode(false) // Debug: surface exceptions
+        .assignmentLogger(::assignmentLogger)
+        .buildAndInitAsync()
+  }
+
+  private fun getResult(
+      assignmentRequest: AssignmentRequest,
+      client: EppoClient,
+      subject: Attributes
+  ): Any {
+    return when (assignmentRequest.assignmentType) {
+      "STRING" ->
+          return client.getStringAssignment(
+              assignmentRequest.flag,
+              assignmentRequest.subjectKey,
+              subject,
+              (assignmentRequest.defaultValue as String))
+
+      "INTEGER" ->
+          return client.getIntegerAssignment(
+              assignmentRequest.flag,
+              assignmentRequest.subjectKey,
+              subject,
+              (assignmentRequest.defaultValue as Int))
+
+      "BOOLEAN" ->
+          return client.getBooleanAssignment(
+              assignmentRequest.flag,
+              assignmentRequest.subjectKey,
+              subject,
+              (assignmentRequest.defaultValue as Boolean))
+
+      "NUMERIC" -> {
+        val defaultNumericValue =
+            if (assignmentRequest.defaultValue is Int) assignmentRequest.defaultValue.toDouble()
+            else (assignmentRequest.defaultValue as Double)
+        return client.getDoubleAssignment(
+            assignmentRequest.flag, assignmentRequest.subjectKey, subject, defaultNumericValue)
+      }
+
+      "JSON" -> {
+        val defaultValue: JsonNode =
+            objectMapper.readTree((assignmentRequest.defaultValue as JSONObject).toString())
+        val jsonNodeResult =
+            client.getJSONAssignment(
+                assignmentRequest.flag, assignmentRequest.subjectKey, subject, defaultValue)
+
+        return jsonNodeResult
+      }
+
+      else -> "NO RESULT"
+    }
+  }
+
+  companion object {
+    private val TAG = TestClientActivity::class.qualifiedName
+    private const val API_KEY = BuildConfig.API_KEY
+    private const val READY_PACKET =
+        "{\"sdkName\":\"example\", \"supportsBandits\" : false, \"sdkType\":\"client\"}"
+    private const val EPPO_API_ADDRESS = "${BuildConfig.EPPO_API_HOST}:${BuildConfig.EPPO_API_PORT}"
+  }
+}
+
+@Composable
+fun TestClientScreen(
+    status: LiveData<String>,
+    assignmentLog: LiveData<String>,
+    modifier: Modifier = Modifier
+) {
+  val statusString: String? by status.observeAsState()
+  val assignmentLogText: String? by assignmentLog.observeAsState()
+
+  Column(modifier = modifier.padding(5.dp)) {
+    Status(statusString ?: "Pending")
+    LogView("Assignment Log", assignmentLogText ?: "Assignment log")
+  }
+}
+
+@Composable
+fun Status(status: String) {
+  Row() {
+    Text("Socket Status:", fontWeight = FontWeight.Bold)
+    Text(status, fontStyle = FontStyle.Italic)
+  }
+}
+
+@Composable
+fun LogView(name: String, log: String) {
+  Column {
+    Text(name, fontWeight = FontWeight.Bold, fontSize = 30.sp)
+    Text(log)
```

Comment:
This is cool! I imagine this lets us see the test logs on the emulated device?

Reply:
yep. There's a bit of UI where the assignments scream by